tools/acrn-crashlog/acrnprobe/conf.rst

```diff
@@ -0,0 +1,379 @@
+      { rank = same; "level 1"; c1; c5;}
+      { rank = same; "level 2"; c2; c4;}
+      { rank = same; "level 3"; c3;}
+
+      node [shape=box;color="transparent";];
+      "None" -> {c1 c5} [ label="inherit 0" ];
+      c1 -> {c2 c4} [ label="inherit 1" ];
+      c2 -> c3 [ label="inherit 2" ];
+   }
+
+Crash type matches at runtime
+=============================
+
+In order to find a more specific type, if one crash type matches
+successfully ``acrnprobe`` will do match for each child of it (if it has)
+continually, and return the last successful one.
+Supposing these crash trees are like the diagram above at runtime:
+If a crash E is triggered, crash E will be returned immediately.
+If a crash A is triggered, then the candidates are crash A, B, C and D.
+The following diagram describes what ``acrnprobe`` will do if the matched
+result is crash D.
+
+.. graphviz::
+
+   digraph {
+      {
+         node [shape=plaintext];
+         "level 1" -> "level 2" -> "level 3";
+      }
+
+      node [shape=box;style="rounded,filled";color=AntiqueWhite;];
+      c1 [ label="crash A\nid 1\ncrash root" ];
+      c2 [ label="crash B\nid 2" ];
+      c3 [ label="crash C\nid 3\ncrash leaf" ];
+      c4 [ label="crash D\nid 4\ncrash leaf" ];
+      { rank = same; "level 1"; c1;}
+      { rank = same; "level 2"; c2; c4;}
+      { rank = same; "level 3"; c3;}
+
+      node [shape=box;style="rounded,dashed";];
+      exp1 [ label="crash B matches fail\nmatch for the next child\nof crash A"];
+      exp2 [ label="crash D matches successfully\nreturn crash D"];
+
+      node [shape=box;style="invis";];
+      "channel" -> c1 [ label="trigger" ]
+      c1 -> {exp1 exp2}
+      exp1 -> c2 -> c3 [ style=dashed dir=none]
+      exp2 -> c4
+   }
+
+Sections
+********
+
+Sender
+======
+
+Example:
+
+.. code-block:: xml
+
+   <sender id="1" enable="true">
+           <name>crashlog</name>
+           <outdir>/var/log/crashlog</outdir>
+           <maxcrashdirs>1000</maxcrashdirs>
+           <maxlines>5000</maxlines>
+           <spacequota>90</spacequota>
+           <uptime>
+                   <name>UPTIME</name>
+                   <frequency>5</frequency>
+                   <eventhours>6</eventhours>
+           </uptime>
+   </sender>
+
+* ``name``:
+  Name of sender. ``Acrnprobe`` uses this label to distinguish different
+  senders.
+  More information about sender, please refer to :ref:`acrnprobe_doc`.
+* ``outdir``:
+  Directory to store generated files of sender. ``Acrnprobe`` will create
+  this directory if it doesn't exist.
+* ``maxcrashdirs``:
+  The maximum serial number of generated ``crash directories``, ``stat directories``
+  and ``vmevent directories``. The serial number will back to 0 if it reaches
+  the maximum. Only used by sender crashlog.
+* ``maxlines``:
+  The maximum lines of file ``history_event``. It will trigger action
+  ``"mv history_event history_event.bak"`` that the lines of ``history_event``
```

Comment:
Change to:

    If the number of lines in the ``history_event`` file reaches the specified ``maxlines``,
    the ``history_event`` file will be renamed to ``history_event.bak`` and logging will
    continue with a now empty ``history_event`` file.